File: app/src/main/java/com/example/uscatterbrain/network/InputStreamConsumer.java
package com.example.uscatterbrain.network;

import io.reactivex.functions.Consumer;

public class InputStreamConsumer extends InputStreamCallback implements Consumer<byte[]> {
    @Override
    public void accept(byte[] bytes) throws Exception {
        if (!closed) {
            acceptBytes(bytes);
        }
    }
}

File: app/src/main/java/net/ballmerlabs/uscatterbrain/network/InputStreamConsumer.kt
package net.ballmerlabs.uscatterbrain.network

import io.reactivex.functions.Consumer
import kotlin.jvm.Throws

class InputStreamConsumer(capacity: Int) : InputStreamCallback(capacity), Consumer<ByteArray?> {
    @Throws(Exception::class)
    override fun accept(bytes: ByteArray?) {
        if (!closed) {
            acceptBytes(bytes!!)
        }
    }
}